packages/sdks/web-component/src/lib/descope-wc/sdkScripts/fingerprint.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  load,
+  defaultEndpoint,
+  defaultScriptUrlPattern,
+} from '@fingerprintjs/fingerprintjs-pro';
+
+export const loadFingerprint = async (
+  initArgs: {
+    publicApiKey: string;
+    useCloudflareIntegration: boolean;
+    cloudflareEndpointUrl: string;
+    cloudflareScriptUrl: string;
+  },
+  _: { baseUrl?: string },
+  onTokenReady: (token: string) => void,
+) => {
+  try {
+    const {
+      publicApiKey,
+      useCloudflareIntegration,
+      cloudflareScriptUrl,
+      cloudflareEndpointUrl,
+    } = initArgs;
+
+    let endpoints = [];
+    if (useCloudflareIntegration && cloudflareEndpointUrl) {
+      const endpointUrl = new URL(cloudflareEndpointUrl);
+      endpoints = [endpointUrl.toString(), defaultEndpoint];
+    } else {
+      endpoints = [defaultEndpoint];
+    }
+
+    let scriptUrlPatterns = [];
+    if (useCloudflareIntegration && cloudflareScriptUrl) {
+      const patterUrl = new URL(cloudflareScriptUrl);
```

Comment:
is it possible that the url already contain query parameters? it may create url that looks like `http://domain.com?k=v?apiKey=<apiKey>&version=<version>&loaderVersion=<loaderVersion>` (twice ?)

Reply:
shouldn't happen, they copy it from their fingerprint console. i don't think it worth handling.